pkg/certificate/providers/vault/certificate_manager.go

```diff
@@ -34,33 +30,28 @@ const (
 	decade                             = 8765 * time.Hour
 )
 
+//todo(schristoff): NewManagement, VaultManager & Certificate.Manager
 // NewCertManager implements certificate.Manager and wraps a Hashi Vault with methods to allow easy certificate issuance.
-func NewCertManager(
-	vaultAddr,
-	token string,
-	role string,
-	cfg configurator.Configurator,
-	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker) (*CertManager, error) {
-	c := &CertManager{
-		role:                        vaultRole(role),
-		cfg:                         cfg,
-		serviceCertValidityDuration: serviceCertValidityDuration,
-		msgBroker:                   msgBroker,
+func NewVaultManager(vaultAddr string, token string, role string) (*CertManager, error) {
+	v := &VaultManager{
+		role: vaultRole(role),
 	}
+
+	c := certificate.NewManager(v)
```

Comment:
the vault manager should only deal with Vault API's, so it won't need to know about the certificate.Manager. We can remove this